core/src/main/clojure/xtdb/expression.clj

```diff
@@ -1375,6 +1375,33 @@
   {:return-type :i32
    :->call-code #(do `(.size ~@%))})
 
+(defmethod codegen-call [:length :utf8] [expr]
+  (codegen-call (assoc expr :f :character_length)))
+
+(defmethod codegen-call [:length :varbinary] [expr]
+  (codegen-call (assoc expr :f :octet_length)))
+
+(defmethod codegen-call [:length :list] [expr]
+  (codegen-call (assoc expr :f :cardinality)))
+
+(defn size-of-set [^Set set] 
+  (.size ^Set set))
+
+(defmethod codegen-call [:length :set] [_] 
```

Comment:
This does pass in a clojure persistent hashset, rather than a xtdb.vector.IListValueReader - can use `count`, though will probably want to investigate the hows/whys of us returning a Set here, rather than a reader.

Reply:
Have moved this over to `count` - leaving comment open RE: types and the like, in case we wish to do more in the future here.